internal/server/http/server.go

```diff
@@ -76,28 +75,31 @@ func (s *Server) Build() error {
 	if _, err := os.Stat(s.impostersPath); os.IsNotExist(err) {
 		return fmt.Errorf("%w: the directory %s doesn't exists", err, s.impostersPath)
 	}
-	var imposterFileCh = make(chan string)
-	var done = make(chan bool)
+	done := make(chan struct{})
 
-	go func() {
-		findImposters(s.impostersPath, imposterFileCh)
-		done <- true
-	}()
-	for {
+	imposterFiles, errc := findImposters(s.impostersPath, done)
+	go s.processImposters(imposterFiles, done)
+
+	if err := <-errc; err != nil {
+		close(done)
+		return fmt.Errorf("%w: failed to find imposters", err)
```

Comment:
instead to finish the up Server, maybe we can only log the error with a user friendly message, what do you think?

Reply:
From my limited knowledge the errors we may expect out of `filepath.Walk` is either `permission denied`, `not exist` or `name too long` I would assume this is such an unusual situation that it should trigger user to check his `ImposterFilePath` because if we silently accept that it might be that some of the imposters were processed while others ignored and we might end up with half walking server. But maybe that is too much. How do you feel about that?

Reply:
So, in this case I like your point of view, maybe is better crash the server in this specific case and inform the user for that error. Ok let try with your approach for now :)